examples/cr3-event-example.c

```diff
@@ -0,0 +1,131 @@
+/* The LibVMI Library is an introspection library that simplifies access to
+ * memory in a target virtual machine or in a file containing a dump of
+ * a system's physical memory.  LibVMI is based on the XenAccess Library.
+ *
+ * Copyright 2011 Sandia Corporation. Under the terms of Contract
+ * DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government
+ * retains certain rights in this software.
+ *
+ * Author: Mathieu Tarral (mathieu.tarral@ssi.gouv.fr)
+ *
+ * This file is part of LibVMI.
+ *
+ * LibVMI is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * LibVMI is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with LibVMI.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/mman.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include <signal.h>
+
+#include <libvmi/libvmi.h>
+#include <libvmi/events.h>
+
+static int interrupted = 0;
+static void close_handler(int sig)
+{
+    interrupted = sig;
+}
+
+event_response_t cr3_callback(vmi_instance_t vmi, vmi_event_t *event)
+{
+    (void)vmi;
+    printf("CR3 write happened: Value=0x%"PRIx64"\n", event->reg_event.value);
+    return VMI_EVENT_RESPONSE_NONE;
+}
+
+int main (int argc, char **argv)
+{
+    vmi_instance_t vmi;
+    status_t status;
+    vmi_mode_t mode;
+    vmi_init_data_t *init_data = alloca(sizeof(vmi_init_data_t)
```

Comment:
I used `alloca` because it was easy to manage for an example code, where we are focused on showing API usage.
And the compiler didn't trigger any warnings, even with `cmake -DHARDENING=ON`.

Should I replace this with a `malloc/free` ?

Reply:
Yes please, the examples should exhibit good coding practices.

Reply:
fixed.